develop/package_basics.Rmd

```diff
@@ -0,0 +1,182 @@
+community and submit it to [CRAN](http://cran.r-project.org).
+
+Tools/Resources
+=========
+
+Recently, many resources have emerged for developing R packages. We will use 
+Hadley Wickham's [R packages](http://r-pkgs.had.co.nz/) for reference, but it's 
+valuable to check out other tutorials.
+
+## Tools
+
+Writing R packages has become much easier with the advent of the IDE
+[Rstudio](http://rstudio.org) and the package
+[devtools](https://github.com/devtools). Using both of these in conjunction will
+make developing your package much easier as they are designed to work together.
+
+- [Download Rstudio here](http://rstudio.org/download)
+
+You can install `devtools` by typing in your R console:
+
+```{r, devtools_install, eval = FALSE}
+install.packages("devtools", repos = "http://cran.rstudio.com/")
+```
+
+You might also need some background noise to avoid distraction. I recommend 
+[the sound of a space ship control room](http://mynoise.net/NoiseMachines/spaceshipNoiseGenerator.php).
+
+## Tutorials
+
+- [Package Structure (Wickham)](http://r-pkgs.had.co.nz/package.html)
+- [Writing an R package from scratch (cat themed) (Parker)](http://hilaryparker.com/2014/04/29/writing-an-r-package-from-scratch/)
+- [Developing R packages (Leek)](https://github.com/jtleek/rpackages)
+- [Creating Packages (Leisch)](http://cran.r-project.org/doc/contrib/Leisch-CreatingPackages.pdf)
+- [Writing R Extensions (CRAN)](http://cran.r-project.org/doc/manuals/r-release/R-exts.html)
+
+Creating your new package
+====================
+
+## What is in an R package?
+
+There are four files/folders needed to create an R package:
+
+- `DESCRIPTION`
+- `R/`
+- `NAMESPACE`
+- `man/`
+
+These are detailed below.
+
+### `DESCRIPTION` 
+
+This file simply lists all the metadata needed for the package. It lists things
+such as Authors, Packages it needs to run, etc. You can view any package's
+DESCRIPTION file by typing `packageDescription("package_name")`
+
+```{r, pkgDesc}
+packageDescription("poppr")
+``` 
+ 
+### `R/`
+
+This is the folder in which all of your `.R` files live. There are different
+philosophies on how to order your files in this folder. Some like to have 
+[a separate R file for each function](https://github.com/dwinter/mmod/tree/master/R), while others like to
+have [all functions in a single R file](https://github.com/liamrevell/Rphylip/tree/master/Rphylip/R). 
+However you want to organize the files in this directory is up to you.
+
+### `NAMESPACE`
+
+This file lists the functions exported by your package (eg `export("myFun")`)
+and functions your package imports from other packages (eg `importFrom("pegas",
+"amova")`). With the modern tools available, you should not need to touch this
+file.
+
+### `man\`
+
+All R packages need documentation. This folder contains all of the documents
+that will become manual pages for your R functions. They are written in a form
+similar to Latex, but completely different. Using the 
```

Comment:
The canonical spelling of LaTeX is using mixed case, so should probably follow that here. (And perhaps hyperlink?)